altmetric/explorer/api/response.py

```diff
@@ -0,0 +1,101 @@
+import requests
+
+
+class Page:
+    '''Encapsulates a page returned from the api and provides accessor methods
+    '''
+
+    def __init__(self, raw_response):
+        '''Initialize a page
+
+        Args:
+            raw_response (requests.Response): the raw response returned by `requests`
+        '''
+        self.raw_response = raw_response
+        self.json = raw_response.json()
+
+    def next_page(self):
+        '''Get the next page if there is one
+
+        The next page is found by looking for the links.next key in an API response.
+        If the key is found then it will contain the URL of the next page and this
+        method will retrieve it and use the response to create and return a new Page.
+
+        If the key is not found then this method will return None.
+
+        Returns:
+            Page: the next page, or None if this is the last page
+        '''
+        url = self.json.get('links', {}).get('next', None)
+        if url:
+            return Page(requests.get(url))
+
+    @property
```

Comment:
What's `@property` doing in this context?

Reply:
That is called a `decorator` and the `@property` decorator allows us to call the function without the `()`.

Reply:
Yup - all good with python decorators, I've just not seen `@property` specifically before.  Is this defined somewhere in the project or is this a standard python library thing?

Reply:
Standard python I believe!

Reply:
Yup.  Standard python https://realpython.com/python-property/